pkg/deploy/api/v1/defaults.go

```diff
@@ -100,8 +100,10 @@ func SetDefaults_RollingDeploymentStrategyParams(obj *RollingDeploymentStrategyP
 func SetDefaults_DeploymentConfig(obj *DeploymentConfig) {
 	for _, t := range obj.Spec.Triggers {
 		if t.ImageChangeParams != nil {
-			t.ImageChangeParams.From.Kind = "ImageStreamTag"
```

Comment:
Oh, that's why we had it unconditionally... I am going to add a comment so I will remember next time:)